inc/zoo/map/robinhood.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "zoo/swar/SWAR.h"
+
+#include <vector>
+
+namespace zoo {
+
+namespace rh {
+
+using u64 = uint64_t;
+using u32 = uint32_t;
+using u16 = uint16_t;
+using u8 = uint8_t;
+
+template<int NBits>
+constexpr auto cheapOkHash(u64 n) {
+    constexpr auto shift = (NBits * ((64 / NBits)-1));
+    constexpr u64 allOnes = meta::BitmaskMaker<u64, 1, NBits>::value;
+    auto temporary = allOnes * n;
+    auto higestNBits = temporary >> shift;
+    return (0 == (64 % NBits)) ? higestNBits : swar::isolate<NBits, u64>(higestNBits);
+}
+
+// The naive method of keeping metadata.
+template<u8 PSLBits, u8 HashBits> struct StrawMetadata {
```

Comment:
Let's not make this a template, it's more useful as a concrete class

Reply:
Sure. Though I wrote this one wrong, I meant to ensure the functions returning psl/hash returned the right width.